File: kotlin-electron/karakum/plugins/convertNodeJsNamespace.js
import ts from "typescript";
import * as karakum from "karakum";

export default function (node, context, render) {
    if (
        ts.isModuleDeclaration(node)
        && ts.isIdentifier(node.name)
        && node.name.text === "NodeJS"
    ) {
        const typeScriptService = context.lookupService(karakum.typeScriptServiceKey)

        return `/* ${typeScriptService?.printNode(node)} */`;
    }
    return null
}

File: kotlin-electron/karakum/src/jsMain/kotlin/electron/karakum/plugins/convertNodeJsNamespace.kt
package electron.karakum.plugins

import electron.karakum.util.nullable
import io.github.sgrishchenko.karakum.extension.createPlugin
import io.github.sgrishchenko.karakum.extension.plugins.TypeScriptService
import io.github.sgrishchenko.karakum.extension.plugins.typeScriptServiceKey
import typescript.isIdentifier
import typescript.isModuleDeclaration

val convertNodeJsNamespace = createPlugin { node, context, _ ->
    nullable {
        ensure(isModuleDeclaration(node))

        val name = node.name
        ensure(isIdentifier(name))
        ensure(name.text == "NodeJS")

        val typeScriptService = ensureNotNull(context.lookupService<TypeScriptService>(typeScriptServiceKey))

        "/* ${typeScriptService.printNode(node)} */"
    }
}
